targets.toml

```diff
@@ -0,0 +1,460 @@
+
+[[target]]
+target = "i686-unknown-linux-gnu"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "native qemu-user qemu-system"
+
+[[target]]
+target = "mips-unknown-linux-gnu"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user"
+
+[[target]]
+target = "mipsel-unknown-linux-gnu"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user qemu-system"
+
+[[target]]
+target = "mips64-unknown-linux-gnuabi64"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+
+[[target]]
+target = "mips64el-unknown-linux-gnuabi64"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user qemu-system"
+
+[[target]]
+target = "mips64-unknown-linux-muslabi64"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user"
+
+[[target]]
+target = "mips64el-unknown-linux-muslabi64"
+os = "ubuntu-latest"
+std = true
+run = true
+
+[[target]]
+target = "powerpc-unknown-linux-gnu"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user qemu-system"
+
+[[target]]
+target = "powerpc64-unknown-linux-gnu"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user qemu-system"
+
+[[target]]
+target = "powerpc64le-unknown-linux-gnu"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user qemu-system"
+
+[[target]]
+target = "riscv64gc-unknown-linux-gnu"
+os = "ubuntu-latest"
+cpp = true
+std = true
+run = true
+runners = "qemu-user qemu-system"
+
+[[target]]
+target = "s390x-unknown-linux-gnu"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user qemu-system"
+
+[[target]]
+target = "sparc64-unknown-linux-gnu"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user qemu-system"
+
+[[target]]
+target = "aarch64-unknown-linux-musl"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+
+[[target]]
+target = "arm-unknown-linux-musleabihf"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+
+[[target]]
+target = "arm-unknown-linux-musleabi"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+
+[[target]]
+target = "armv5te-unknown-linux-gnueabi"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+
+[[target]]
+target = "armv5te-unknown-linux-musleabi"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+
+[[target]]
+target = "armv7-unknown-linux-musleabi"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+
+[[target]]
+target = "armv7-unknown-linux-musleabihf"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+
+[[target]]
+target = "i586-unknown-linux-musl"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user"
```

Comment:
Should we also allow a native runner here? It needs `qemu-user` to ensure we use the right CPU, but it can also run natively (for whatever that's worth). It's also the default when running via `cross`, so it's probably a good idea.

Reply:
added this, we're already testing it so nothing has been missed but good to be explicit imo